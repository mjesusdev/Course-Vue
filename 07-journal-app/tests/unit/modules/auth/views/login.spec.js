import { shallowMount } from '@vue/test-utils'
import Login from '@/modules/auth/views/login.vue'

import createVuexStore from '../../../mock-data/mock-store'

import Swal from 'sweetalert2'
jest.mock('sweetalert2', () => ({
    fire: jest.fn(),
    showLoading: jest.fn(),
    close: jest.fn()
}))

describe('Tests in Login Component', () => {

    const store = createVuexStore({
        status: 'not-authenticated',
        user: null,
        idToken: null,
        refreshToken: null
    })

    store.dispatch = jest.fn()

    beforeEach(() => jest.clearAllMocks())

    test('should do match with SnapShot', () => {
        const wrapper = shallowMount(Login, {
            global: {
                plugins: [ store ]
            }
        })
        expect( wrapper.html() ).toMatchSnapshot()
    })

    test('user or password wrong', async() => {
        store.dispatch.mockReturnValueOnce({ ok: false, message: 'Error in credentials' })

        const wrapper = shallowMount(Login, {
            global: {
                plugins: [ store ]
            }
        })
        
        await wrapper.find('form').trigger('submit')
        expect( store.dispatch ).toHaveBeenCalledWith('auth/signInUser', { email: '', password: '' })
        expect( Swal.fire ).toHaveBeenCalledWith('Error', 'Error in credentials', 'error')
    })
    
    test('should redirect to route no-entry', async() => {
        store.dispatch.mockReturnValueOnce({ ok: true })
    
        const wrapper = shallowMount(Login, {
            global: {
                plugins: [ store ]
            }
        })

        const [ txtEmail, txtPassword ] = wrapper.findAll('input')
        await txtEmail.setValue('pepe@gmail.com')
        await txtPassword.setValue('123456')

        await wrapper.find('form').trigger('submit')

        expect( store.dispatch ).toHaveBeenCalledWith('auth/signInUser', { email: 'pepe@gmail.com', password: '123456' })
        expect( wrapper.router.push ).toHaveBeenCalledWith({ 'name': 'no-entry' })
    })
})