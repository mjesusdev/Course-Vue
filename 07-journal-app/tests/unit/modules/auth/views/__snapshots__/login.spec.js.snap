// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Login Component should do match with SnapShot 1`] = `
<div><span class="login100-form-title p-b-41"> Login </span>
  <form class="login100-form validate-form p-b-33 p-t-5">
    <div class="wrap-input100 validate-input" data-validate="Enter username"><input class="input100" type="text" placeholder="Email" required=""><span class="focus-input100" data-placeholder=""></span></div>
    <div class="wrap-input100 validate-input" data-validate="Enter password"><input class="input100" type="password" placeholder="Password" required=""><span class="focus-input100" data-placeholder=""></span></div>
    <div class="container-login100-form-btn m-t-32"><button class="login100-form-btn"> Login </button></div>
    <div class="container-login100-form-btn m-t-32">
      <router-link-stub to="[object Object]" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
    </div>
  </form>
</div>
`;
